back/app/models/user.rb

```diff
@@ -500,15 +513,39 @@ def increment_confirmation_retry_count!
     save!
   end
 
-  def reset_email!(email)
-    update!(
-      email: email,
-      email_confirmation_code_reset_count: 0
-    )
+  def reset_email!(new_email)
+    if user_confirmation_enabled? && active?
+      update!(new_email: new_email, email_confirmation_code_reset_count: 0)
+    else
```

Comment:
We will also enter this branch if confirmation is enabled but the user is not active. Is this intentional?

Reply:
Yes it's intentional, but the validation will return errors if the update is rejected